### PR TITLE
Bug fix for 3D Group Conv Backward data and weight update. Failure noticed when pads and strides are not 1

### PR DESCRIPTION
**Pull request by amberhassaan**
3D Group Convolution Backward datd and weights update solvers had a bug. Initially it was thought the bug was in CK but in reality it was in MIOpen. 
The bug stems from a design decision in MIOpen that swaps input and output tensors in backward passes. This is very confusing and bug-prone and needs to be fixed at large ASAP.

### CODE REVIEW

**Comment by junliume** [comment]
@amberhassaan could you submit an issue at https://github.com/ROCmSoftwarePlatform/composable_kernel/issues

File: test/gtest/group_conv3d_bwd.hpp [hunk 1]
```diff
@@ -30,6 +30,7 @@
 std::vector<Conv3DTestCase> ConvTestConfigs()
 { // g    n   c   d    h   w   k   z  y  x pad_x pad_y pad_z stri_x stri_y stri_z dia_x dia_y dia_z
     return {{1, 128, 64, 14, 28, 28, 64, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {2, 128, 32, 28, 28, 28, 32, 3, 3, 3, 0, 0, 0, 2, 2, 2, 1, 1, 1, miopenConvolution},
```
Comment (amberhassaan) [inline test/gtest/group_conv3d_bwd.hpp]: This row of inputs causes numerical failure.

File: test/gtest/group_conv3d_wrw.hpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 std::vector<Conv3DTestCase> ConvTestConfigs()
 { // g   n   c   d    h   w   k   z  y  x pad_x pad_y pad_z stri_x stri_y stri_z dia_x dia_y dia_z
     return {{1, 128, 64, 14, 28, 28, 64, 3, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, miopenConvolution},
+            {2, 128, 32, 28, 28, 28, 32, 3, 3, 3, 0, 0, 0, 2, 2, 2, 1, 1, 1, miopenConvolution},
```
Comment (amberhassaan) [inline test/gtest/group_conv3d_wrw.hpp]: This row causes a very long execution time or a hang. Not sure. If I reduce the `N` and `C, K` values, I see a numerical failure just like `bwd` pass.

**Comment by amberhassaan** [comment]
To reproduce failure, build MIOpen in debug mode
```
make -j 16 test_group_conv3d_bwd test_group_conv3d_wrw
BUILD_DIR/bin/test_group_conv3d_bwd
BUILD_DIR/bin/test_group_conv3d_wrw
```

**Comment by amberhassaan** [comment]
@junliume , @JehandadKhan : Please approve and merge asap. This is needed for #2429 and provides a bug fix in general (something that previous tests didn't pick up).

**Comment by atamazov** [comment]
@JehandadKhan @junliume The root reason of the issue is insufficient testing of the new solvers.

Other reasons:
- Insufficient/inaccurate reviewing
  - One of the rare cases when this kind of problem was detected by the reviewer is [here](https://github.com/ROCm/MIOpen/pull/2531#discussion_r1409784681)
  - The number of experienced reviewers of #2373 was clearly insufficient. Some of the reviewers were invited but didn't provide any feedback.
- The approach to the naming of tensors in convolutions is confusing.
  - [Informative] For forward x tensor is a read-only "input", y is writable "output"; for backward, dy is read-only "input", and x is writable "output".
    - **_While it may seem logical for some programmers (who invented this stuff), it confuses the most of others ;) And it does not work for WrW convolutions anyway!_**
  - [Informative] In some old solvers "bottom" and "top" are used (for forward, x is "bottom", y is "top"; for backward, dy is "bottom", dx is "top")
- New programmers are usually unaware of the legacy naming approach.
- A large amount of work to refactor with a seemingly small result (which is why the rework of this stuff was constantly postponed)

### Proposal: let's use "x", "y" and "w" (and their derivatives, e.g. "dxDesc" etc) everywhere.

/cc @averinevg @DrizztDoUrden

**Comment by amberhassaan** [comment]
I agree with the proposal above wholeheartedly. This has been a source of confusing and has taken up multiple developer hours multiple times. 
@atamazov : please create an issue and I will take it up after the deadline.

**Review by iq136boy — APPROVED** [review]

**Review by averinevg — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**